### ISSUE TITLE
Looks awesome!

### ISSUE THREAD

**Issue by gabesullice** [cooperdow/Hangman#8]
Hey guys,

@mrmikemcguire told me you guys had all finished your projects. I checked them all out and they all look really cool. I'm super impressed (and a bit jealous) that all you guys were able to do this in HS and in a trimester! You're putting me to shame.

After I saw your hangman project, I got really interested the idea of having code try to guess an unknown word. Specifically, how to efficiently deal with a huge number of words and filter out huge swathes of possible words really quickly. So, I got the itch and I tried to do it myself.

I never actually put together a front-end for it, but I did implement a trie (I think I mentioned it to you as a possible way to deal with all the words). I know this was just a school project that you're done with now, but if you're interested in this stuff and want to flip your game upside down by having the computer guess a word, I made this to help: https://github.com/gabesullice/trie.js.

Basically, it's a little library that you can use to take a huge list of words, read it into code, and efficiently start hacking away possibilities until you're left with only one possible word. I toyed around with it (see app.js) and some interesting things came up. It can find the word "encyclopedia" in two guesses, but much simpler words can take more than 10. It might be fun to play around and see if you can find a word that it can guess in 1 try or which word(s) take the most guesses (I haven't tried either).

Keep up that awesome work!

Best,

Gabe
